pkg/util/exithandler.go

```diff
@@ -28,22 +28,26 @@ func RegisterExitHandler(handler func()) {
 // SetupExitSignalHandling sets up a signal channel to listen for termination or interruption signals.
 // When one of these signals is received, all the registered exit handlers will be invoked, just
 // before terminating the program.
-func SetupExitSignalHandling() {
-	setupExitSignalHandling(true)
+func SetupExitSignalHandling(exitChan chan bool) {
+	setupExitSignalHandling(exitChan, true)
 }
 
 // setupExitSignalHandling sets up a signal channel to listen for termination or interruption signals.
 // When one of these signals is received, all the registered exit handlers will be invoked.
 // It is possible to prevent the program from exiting by setting the doExit param to false (used for testing)
-func setupExitSignalHandling(doExit bool) {
+func setupExitSignalHandling(exitChan chan bool, doExit bool) {
 	sigChan := make(chan os.Signal, 2)
 	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
 	go func() {
+		defer func() {
+			signal.Stop(sigChan)
+			close(exitChan)
+		}()
 		for sig := range sigChan {
 			log.Printf("captured %v, calling exit handlers and exiting..", sig)
 			ExecuteExitHandlers()
 			if doExit {
```

Comment:
I think `doExit` is now redundant, you can pass an `exitChan` in the tests, but never try to read from it. Or you can check something is available in `exitChan` after you sent a signal.